### PR TITLE
Weather Immunity Refactor

### LINKED ISSUE #17646: Sentient necropolis tendrils take damage from ash storms
They also get attacked by lavaland mobs.

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
No longer uses mob factions, but checks against a list of their weather communities instead.

It's like faction checks but for weather, featuring MAGIC STRINGS which I'll make defines when I wake up I guess.

Also made it so sentience potions won't work on mob spawners.

Fixes #17646

### CODE REVIEW

File: code/modules/mob/living/simple_animal/spawner.dm [hunk 1]
```diff
@@ -19,6 +19,7 @@
 	minbodytemp = 0
 	maxbodytemp = 350
 	layer = MOB_LAYER-0.1
+	sentience_type = SENTIENCE_BOSS
```
Comment (optimumtact) [inline code/modules/mob/living/simple_animal/spawner.dm]: hey

**Comment by Iamgoofball** [comment]
idea:
if you dont have a connected client, storm does no damage

**Comment by RandomMarine** [comment]
People would purposely close their client to avoid storms in a pinch then.

**Comment by Iamgoofball** [comment]
leaving you vulnerable to everything else on lavaland?

**Comment by Armhulen** [comment]
Yep! There is some areas you cleared of enemies I hope.

**Comment by Razharas** [comment]
This really should be a proc so we can have more advanced conditions for immunity

File: code/modules/mining/lavaland/weather.dm [hunk 2]
```diff
@@ -73,9 +74,8 @@
 
 
 /datum/weather/proc/storm_act(mob/living/L)
-	if(prob(30)) //Dont want it spammed very tick
-		L << "You're buffeted by the storm!"
-		L.adjustBruteLoss(1)
+	if(immunity_type in L.weather_immunities)
+		return
```
Comment (optimumtact) [inline code/modules/mining/lavaland/weather.dm]: you've removed the effect entirely?

Comment (KorPhaeron) [inline code/modules/mining/lavaland/weather.dm]: Not a real weather type anyway, just the example

**Pull request merged**